crates/sui-bridge/src/config.rs

```diff
@@ -0,0 +1,267 @@
+            Some(modules) => {
+                if modules.is_empty() {
+                    return Err(anyhow!(
+                        "`sui_bridge_modules` is required when `run_client` is true"
+                    ));
+                }
+                modules
+                    .iter()
+                    .map(|module| Identifier::from_str(module))
+                    .collect::<Result<Vec<_>, _>>()
+                    .map_err(|e| anyhow!("Error parsing sui module: {:?}", e))?
+            }
+            None => {
+                return Err(anyhow!(
+                    "`sui_bridge_modules` is required when `run_client` is true"
+                ))
+            }
+        };
+
+        let mut sui_bridge_modules_start_tx_override = BTreeMap::new();
+        match &self.sui_bridge_modules_start_tx_override {
+            Some(overrides) => {
+                for (module, tx_digest) in overrides {
+                    let module = Identifier::from_str(module)?;
+                    if sui_bridge_modules.contains(&module) {
+                        let tx_digest = TransactionDigest::from_str(tx_digest)?;
+                        sui_bridge_modules_start_tx_override.insert(module, tx_digest);
+                    } else {
+                        return Err(anyhow!(
+                            "Override start tx digest for module {:?} is not in `sui_bridge_modules`",
+                            module
+                        ));
+                    }
+                }
+            }
+            None => {}
+        }
+
+        let (gas_coin, gas_object_ref, owner) = sui_client
+            .get_gas_data_panic_if_not_gas(gas_object_id)
+            .await;
+        if owner != Owner::AddressOwner(client_sui_address) {
+            return Err(anyhow!("Gas object {:?} is not owned by bridge client key's associated sui address {:?}, but {:?}", gas_object_id, client_sui_address, owner));
+        }
+        info!(
+            "Starting bridge client with gas object {:?}, balance: {}",
+            gas_object_ref.0,
+            gas_coin.value()
+        );
+        let bridge_client_config = BridgeClientConfig {
+            sui_address: client_sui_address,
+            key: bridge_client_key,
+            gas_object_ref,
+            metrics_port: self.metrics_port,
+            sui_client: sui_client.clone(),
+            eth_client: eth_client.clone(),
+            db_path,
+            eth_addresses,
+            sui_bridge_modules,
+            eth_addresses_start_block_number_override,
+            sui_bridge_modules_start_tx_override,
+        };
+
+        Ok((bridge_server_config, Some(bridge_client_config)))
+    }
+}
+
+pub struct BridgeServerConfig {
+    pub key: BridgeAuthorityKeyPair,
+    pub server_listen_port: u16,
+    pub metrics_port: u16,
+    pub sui_client: Arc<SuiClient<SuiSdkClient>>,
+    pub eth_client: Arc<EthClient<ethers::providers::Http>>,
+}
+
+// TODO: add gas balance alert threshold
```

Comment:
It seems like you have to restart the server with a new gas object if gas was to run out, would it make the ux better if we just queried the address for available gas coins that can be used. Then we someone could just send the sui address more gas when needed without restarting the client?

Reply:
it would certainly work and potentially provide better ux. We can add this in the future, keep this simple for now